app/controllers/logs_controller.rb

```diff
@@ -1,14 +1,46 @@
 class LogsController < ApplicationController
   def index
-    logs = UseCases::Administrator::GetAuthRequests.new(
+    if location_outside_organisation?
+      redirect_to(location_new_logs_search_path) && return
+    end
+
+    if valid_search_params.blank?
+      redirect_to(new_logs_search_path) && return
+    end
+
+    @location_name = Location.find_by(id: params[:location])&.address
+
+    @logs = get_auth_requests
```

Comment:
No biggie, but given this is only called in one place I would generally to not pull this out into a method.  I always try only do restful methods in controllers

Reply:
Yeah, I went back and forth on this, it looked messy in-place. Gonna look at it again

RESTful-only in controller is interesting, I'm definitely happy to use arbitrary private methods. I can think of some reasons to avoid extra methods, but interested to hear your reasons?

Reply:
Still interested to hear your opinions, but think this is addressed in https://github.com/alphagov/govwifi-admin/pull/404/commits/c83117a178aa650da4afb7cd71a14752f90a259a

Reply:
And as I played with it more, I actually ended up putting this back in (https://github.com/alphagov/govwifi-admin/pull/404/commits/f37358b5b0caa9e5a2dfc435f6f6c9d05af04596). I think the use case's composition isn't the most important thing to a reader, so sinking it down one level of indirection feels right.